conn.go

```diff
@@ -429,37 +433,37 @@ func padToBlockSize(payload []byte, blockSize int) (prefix, finalBlock []byte) {
 }
 
 // encrypt encrypts and macs the data in b.
-func (hc *halfConn) encrypt(b *block, explicitIVLen int) (bool, alert) {
+func (hc *halfConn) encrypt(t byte, b *block, explicitIVLen int) (bool, alert) {
```

Comment:
Note to self: this function got out of control, the abstraction is fully broken, it might as well make the header itself. For future refactor.